src/ott/tools/sliced.py

```diff
@@ -0,0 +1,75 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from ott.geometry import costs, pointcloud
+from ott.solvers import linear
+
+__all__ = ["sliced_w"]
+
+Randfunc_t = Callable[[jax.Array, (int, int)], jnp.ndarray]
+
+
+def sliced_w(
+    x: jnp.ndarray,
+    y: jnp.ndarray,
+    a: jnp.ndarray,
+    b: jnp.ndarray,
+    cost_fn: Optional[costs.CostFn] = None,
+    proj_matrix: Optional[jnp.ndarray] = None,
+    n_proj: int = 100,
+    random_direction_generator: Optional[Randfunc_t] = None,
+    rng: Optional[jax.Array] = None,
```

Comment:
I'd add option to return the univariate output as well, wdyt?

Reply:
yes, why not, but then we need to pass kwargs on whether to return dual potentials etc...